tests/suites/os/tests/config-json/index.js

```diff
@@ -26,53 +71,21 @@ module.exports = {
 
 				const context = this.context.get();
 
-				test.comment(`Waiting for os-config-json service to be inactive...`);
-				return context.systemd.waitForServiceState(
-						'os-config-json.service',
-						'inactive',
-						context.link
-				).then(() => {
-					test.comment(`Setting hostname to ${hostname} in config.json...`);
-					return context.worker.executeCommandInHostOS(
-						[
-							`tmp=$(mktemp)`,
-							`&&`, `cat`,  `/mnt/boot/config.json`,
-							`|`, `jq`, `'.hostname="${hostname}"'`,
-							`>`, `$tmp`,
-							`&&`, `mv`, `"$tmp"`, `/mnt/boot/config.json`,
-						].join(' '),
-						context.link);
-				}).then(() => {
-					return context.systemd.waitForServiceState(
-						'avahi-daemon.service',
-						'active',
-						`${hostname}.local`,
-					)
-				}).then(() => {
-					return context.worker.executeCommandInHostOS(
-						'cat /etc/hostname',
-						`${hostname}.local`
-					)
-				}).then((actual) => {
-					const expected = hostname;
-					test.equal(actual, expected, 'Device should have a new hostname');
-				}).then(() => {
-					// Remove hostname
-					return context.worker.executeCommandInHostOS(
-						[
-							`tmp=$(mktemp)`,
-							`&&`, `jq`, `"del(.hostname)"`, `/mnt/boot/config.json`,
-							`>`, `$tmp`, `&&`, `mv`, `"$tmp"`, `/mnt/boot/config.json`
-						].join(' '),
-						`${hostname}.local`,
+				return setConfig(test, context, 'hostname', hostname) 
```

Comment:
setConfig() also seems something the framework could provide to use in all tests

Reply:
setConfig() is only applicable to balenaOS so I don't see it fitting in the leviathan framework.

Do you mean at the top of the os suite?

Reply:
The pattern is that a test from any suite may want to configure the OS, that's why I said leviathan. I could see leviathan then checking the OS and doing different things. Even if we had a SV endpoint for this, tests might not want to assume the SV is running, or might just want to speed things up by direct configuration.

It's just a comment as I find that we keep duplicating code and introducing the same errors.